port/raspberrypi/rp2xxx/src/hal/uart.zig

```diff
@@ -367,14 +400,21 @@ pub const UART = enum(u1) {
     }
 
     /// Convenience function for waiting for a single byte to come across the RX line.
-    pub fn read_word(uart: UART, timeout: ?mdf.time.Duration) ReceiveError!u8 {
+    pub fn read_word_blocking(uart: UART, timeout: ?mdf.time.Duration) ReceiveBlockingError!u8 {
         var byte: [1]u8 = undefined;
         try uart.read_blocking(&byte, timeout);
         return byte[0];
     }
 
+    /// Read a single byte from the RX line if available otherwise returns
+    /// `error.WouldBlock`.
+    pub fn read_word(uart: UART) ReceiveNonBlockingError!u8 {
+        if (!uart.is_readable()) return ReceiveNonBlockingError.WouldBlock;
+        return try uart.read_rx_fifo_with_error_check();
+    }
```

Comment:
Why not just make that function be `ReceiveError!?u8` instead of `error.WouldBlock`?

Then you can do something like this:
```zig
while(try uart.read_word()) |word| {
    …
}
```